main/src/cgeo/geocaching/models/WaypointUserNoteCombiner.java

```diff
@@ -0,0 +1,84 @@
+package cgeo.geocaching.models;
+
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class WaypointUserNoteCombiner {
+
+    private static final String SEPARATOR = "\n--\n";
+    private static final Pattern PATTERN_SEPARATOR_SPLIT = Pattern.compile("\\s*" + SEPARATOR + "\\s*");
+
+    private Waypoint waypoint;
+
+    /**
+     *
+     */
+    public WaypointUserNoteCombiner(final Waypoint waypoint) {
+        this.waypoint = waypoint;
+    }
+
+
+    /**
+     * Combine note and user note. Separated with Separator "\n--\n"
+     *
+     * @return string with combined note
+     */
+    public final String getCombinedNoteAndUserNote() {
+        final String userNote = this.waypoint.getUserNote();
+        if (this.waypoint.isUserDefined()) {
+            return userNote;
+        } else {
+            final StringBuilder newNote = new StringBuilder(this.waypoint.getNote());
+            if (StringUtils.isNotEmpty(userNote)) {
+                newNote.append(SEPARATOR);
+                newNote.append(userNote);
+            }
+            return newNote.toString();
+        }
+    }
+
+    /**
+     * Split up given string into note and user note (separated with \n--\n).
+     * For userDefined waypoints only userNote is set.
+     *
+     * @param combinedNote note to split up
+     */
+    public void updateNoteAndUserNote(final String combinedNote) {
+        if (combinedNote != null) {
+            String fixedCombinedNote = combinedNote;
+            // \n was removed via validate
+            if (combinedNote.startsWith("--\n")) {
```

Comment:
Use the constant here instead of hardcoded --

Reply:
Done